checks/automatic_dependency_update.go

```diff
@@ -20,32 +20,38 @@ import (
 	"github.com/ossf/scorecard/checker"
 )
 
-const CheckAutomaticDependencyUpdate = "Automatic-Dependency-Update"
+const checkAutomaticDependencyUpdate = "Automatic-Dependency-Update"
 
 //nolint
 func init() {
-	registerCheck(CheckAutomaticDependencyUpdate, AutomaticDependencyUpdate)
+	registerCheck(checkAutomaticDependencyUpdate, AutomaticDependencyUpdate)
 }
 
 // AutomaticDependencyUpdate will check the repository if it contains Automatic dependency update.
 func AutomaticDependencyUpdate(c *checker.CheckRequest) checker.CheckResult {
-	result := CheckIfFileExists(CheckAutomaticDependencyUpdate, c, fileExists)
-	if !result.Pass {
-		result.Confidence = 3
+	r, err := CheckIfFileExists2(checkAutomaticDependencyUpdate, c, fileExists)
+	if err != nil {
+		return checker.MakeInconclusiveResult(checkAutomaticDependencyUpdate, err)
 	}
-	return result
+	if !r {
+		c.CLogger.Fail("E01", "no configuration file found in the repo")
+		return checker.MakeInconclusiveResult(checkAutomaticDependencyUpdate, nil)
```

Comment:
I'm thinking of updating the signature of `MakeInconclusiveResult ` to take a `CLogger` and code number as input: this way we can simply call `MakeInconclusiveResult ` and get a code number added to the check details. That will reduce cognitive load on our side.